lib/cluster/cpg.c

```diff
@@ -757,6 +757,24 @@ pcmk_cpg_membership(cpg_handle_t handle,
     counter++;
 }
 
+/*!
+ * \brief Set the CPG deliver callback function for a cluster object
+ *
+ * \param[in,out] cluster  Cluster object
+ * \param[in]     fn       Deliver function to set
+ *
+ * \return Standard Pacemaker return code
```

Comment:
mention that calls must be guarded with SUPPORT_COROSYNC

Reply:
This is in `cpg.c`, seems implied

Reply:
External API users don't know that

Reply:
Either:
* They found it via cluster.h, which guards the declaration within SUPPORT_COROYNC, or
* They didn't find it at all, because functions guarded by SUPPORT_COROSYNC don't show up in the HTML docs.

In any case, all functions in cpg.c and corosync.c should be documented as requiring SUPPORT_COROSYNC if we flag this one. Those source files aren't compiled, and their functions not declared, unless SUPPORT_COROSYNC is true.

Reply:
Technically BUILD_CS something or other but same idea

Reply:
> * They didn't find it at all, because functions guarded by SUPPORT_COROSYNC don't show up in the HTML docs.

I just realized libcrmcluster isn't documented in doxygen at all (no `\defgroup` for it in doxygen.h, plus no `\ingroup` in cluster.h) :-/

But once that's fixed, that should be the definitive entry point for external developers, and it wouldn't have any indication about `SUPPORT_COROSYNC`.

> all functions in cpg.c and corosync.c should be documented as requiring SUPPORT_COROSYNC if we flag this one.

Yeah, don't worry about it, it can be its own project

Reply:
https://projects.clusterlabs.org/T818